src/components/Button/Button.test.tsx

```diff
@@ -0,0 +1,118 @@
+import userEvent from '@testing-library/user-event';
+import { describe, it, expect, vi } from 'vitest';
+
+import { render, screen } from '../../test/test-utils';
+
+import { Button } from './Button';
+
+describe('Button', () => {
+  it('renderiza con props por defecto', () => {
+    render(<Button data-testid="btn">Hola</Button>);
+    const btn = screen.getByTestId('btn');
+    expect(btn).toBeInTheDocument();
+    expect(btn).toHaveTextContent('Hola');
+    expect(btn).not.toBeDisabled();
+    expect(btn).toHaveAttribute('type', 'button'); // default
+  });
+
+  it('dispara onClick cuando está habilitado', async () => {
+    const onClick = vi.fn();
+    render(
+      <Button data-testid="btn" onClick={onClick}>
+        Click
+      </Button>
+    );
+    await userEvent.click(screen.getByTestId('btn'));
+    expect(onClick).toHaveBeenCalledTimes(1);
+  });
+
+  it('no dispara onClick si está disabled', async () => {
+    const onClick = vi.fn();
+    render(
+      <Button data-testid="btn" disabled onClick={onClick}>
+        Click
+      </Button>
+    );
+    const btn = screen.getByTestId('btn');
+    expect(btn).toBeDisabled();
+    await userEvent.click(btn);
+    expect(onClick).not.toHaveBeenCalled();
+  });
+
+  it('muestra spinner y bloquea interacción en loading', async () => {
```

Comment:
Test name is Spanish ('muestra spinner y bloquea interacción en loading'). Also the test asserts that the button is disabled; prefer asserting behavior through roles and visible spinner semantics.

**Sugerencia:**
```typescript
Use: it('shows spinner and blocks interaction when loading', async () => { const user = userEvent.setup(); ... expect(screen.getByRole('progressbar')).toBeInTheDocument(); await user.click(screen.getByRole('button')); expect(onClick).not.toHaveBeenCalled(); });
```